subprojects/veins_catch/src/SignalRepresentation.cc

```diff
@@ -0,0 +1,954 @@
+                                REQUIRE(signal2.getSendingStart() == signal3.getSendingStart());
+                                REQUIRE(signal2.getDuration() == signal3.getDuration());
+                                REQUIRE(signal2.getPropagationDelay() == signal3.getPropagationDelay());
+
+                                REQUIRE(signal2.getNumAnalogueModelsApplied() == signal3.getNumAnalogueModelsApplied());
+                                REQUIRE(signal2.getBitrate() == signal3.getBitrate());
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+SCENARIO("Signal Value Access", "[toolbox]")
+{
+    DummySimulation ds(new cNullEnvir(0, nullptr, nullptr)); // necessary so simtime_t works
+    GIVEN("A spectrum with frequencies (1,2,3,4,5,6)")
+    {
+        Freqs freqs = {1, 2, 3, 4, 5, 6};
+
+        SpectrumPtr spectrum = Spectrum::getInstance(freqs);
+
+        WHEN("a signal is constructed and values assigned to frequencies")
+        {
+            Signal signal(spectrum);
+            signal[1] = 4;
+            signal[2] = 3;
+            signal[3] = 2;
+            signal[4] = 1;
+
+            // Access via operators
+            THEN("relative interval is as expected")
+            {
+                REQUIRE(signal.getNumAbsoluteValues() == 6);
+
+                REQUIRE(signal.getRelativeOffset() == 1);
+                REQUIRE(signal.getNumRelativeValues() == 4);
+
+                REQUIRE(signal.getRelativeStart() == 1);
+                REQUIRE(signal.getRelativeEnd() == 5); // Note: This is not 4 as it should support iterations (what about <= in for-loop?)
+            }
+            THEN("frequencies can be addressed by index")
+            {
+                REQUIRE(signal[1] == 4);
+                REQUIRE(signal[2] == 3);
+                REQUIRE(signal[3] == 2);
+                REQUIRE(signal[4] == 1);
+            }
+            THEN("frequencies can be addressed by frequency")
+            {
+                REQUIRE(signal(2) == 4);
+                REQUIRE(signal(3) == 3);
+                REQUIRE(signal(4) == 2);
+                REQUIRE(signal(5) == 1);
+            }
+
+            WHEN("data interval defined")
+            {
+                signal.setDataStart(2);
+                signal.setDataEnd(3);
+
+                THEN("iterating over data interval returns correct values")
+                {
+                    for (uint16_t i = signal.getDataStart(); i < signal.getDataEnd(); i++)
+                        REQUIRE(signal[i] == 5 - i);
+                }
+            }
+        }
+    }
+}
+
+SCENARIO("Signal Timing", "[toolbox]")
+{
+    DummySimulation ds(new cNullEnvir(0, nullptr, nullptr)); // necessary so simtime_t works
+    GIVEN("A spectrum with frequencies (1,2,3,4) and a signal (4,3,2,1)")
+    {
+        Freqs freqs = {1, 2, 3, 4};
```

Comment:
This is most likely too small for the `signal[4]` that is written to later on.
```suggestion
        Freqs freqs = {1, 2, 3, 4, 5, 6};
```

Reply:
Thanks @sommer for the hint. This actually lead to the root of a bigger problem: Signal allowed silent out-of-bound access to the underlying array. I'm adding some asserts and tests for that.